caikit/runtime/train.py

```diff
@@ -0,0 +1,198 @@
+
+    # Import libraries to register modules
+    for library in args.library or []:
+        log.info("<COR88091092I>", "Importing library %s", library)
+        importlib.import_module(library)
+
+    # Try to import the root library of the provided module. It's ok if this
+    # fails since the module may be a UID
+    mod_root_lib = args.module.split(".")[0]
+    try:
+        importlib.import_module(mod_root_lib)
+    except ImportError:
+        log.debug("Unable to import module root lib: %s", mod_root_lib)
+
+    # Figure out the module to train
+    mod_reg = module_registry()
+    mod_pkg_to_mod = {
+        f"{mod.__module__}.{mod.__name__}": mod for mod in mod_reg.values()
+    }
+    module: Type[ModuleBase] = mod_reg.get(args.module, mod_pkg_to_mod.get(args.module))
+    error.value_check(
+        "<COR03876205E>",
+        module is not None,
+        "Unable to find module {} to train",
+        args.module,
+    )
+
+    # Read training kwargs
+    try:
+        if os.path.isfile(args.training_kwargs):
+            with open(args.training_kwargs, encoding="utf-8") as handle:
+                training_kwargs = json.load(handle)
+        else:
+            training_kwargs = json.loads(args.training_kwargs)
+    except json.decoder.JSONDecodeError:
+        log.error(
+            "<COR65834760E>",
+            "--training-kwargs must be valid json or point to a valid json file",
+        )
+        raise
+
+    # Convert datatypes to match the training API
+    training_service = ServicePackageFactory.get_service_package(
+        ServicePackageFactory.ServiceType.TRAINING,
+    )
+    train_rpcs = [
+        rpc
+        for rpc in training_service.caikit_rpcs.values()
+        if rpc.module_list == [module]
+    ]
+    error.value_check(
+        "<COR11978965E>",
+        len(train_rpcs) == 1,
+        "Unable to find a unique train signature",
+    )
+    package_name = get_service_package_name(ServicePackageFactory.ServiceType.TRAINING)
+    train_rpc_req = (
+        train_rpcs[0].create_request_data_model(package_name).get_proto_class()
+    )
+    request_proto = json_format.Parse(
+        json.dumps({"parameters": training_kwargs}),
+        train_rpc_req(),
+    )
+    req_kwargs = build_caikit_library_request_dict(
+        request_proto.parameters, module.TRAIN_SIGNATURE
+    )
+    train_kwargs.update(req_kwargs)
+    log.debug3("All train kwargs: %s", train_kwargs)
+
+    # Run the training
+    with alog.ContextTimer(
+        log.info,
+        "Finished training %s in: ",
+        args.model_name,
+    ):
+        future = train(module, wait=True, **train_kwargs)
+        info = future.get_info()
+        if info.status == TrainingStatus.COMPLETED:
+            log.info("Training finished successfully")
+            return 0
+        else:
+            log.error("Training finished unsuccessfully")
+            for err in info.errors or []:
+                log.error(err)
+            return 1
```

Comment:
concerned about the error handling here. In case of validation or runtime errors, currently the model's train function with throw error, which then will get rethrown ?

Reply:
Is there a mechanism to surface the actual error to user in the script consuming this script?

Reply:
Good question. Right now, no, there is no better mechanism beyond looking at the logs. Assuming that this is the `command` to the `Job` that eventually gets run, I'm not sure there's any more sophisticated mechanism to report errors beyond logging them, right? I suppose we could/should see if it's possible to get the full stack trace out of that `log.error(err)` line, though